app/src/main/java/com/example/nerdlauncher/ActivityAdapter.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.nerdlauncher
+
+import android.content.Intent
+import android.content.pm.ResolveInfo
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class ActivityAdapter(private val activities: List<ResolveInfo>) :
+    RecyclerView.Adapter<ActivityAdapter.ActivityHolder>() {
+
+    class ActivityHolder(itemView: View) :
+        RecyclerView.ViewHolder(itemView),
+        View.OnClickListener {
+        private val nameTextView = itemView as TextView
+        private lateinit var resolveInfo: ResolveInfo
+
+        init {
+            nameTextView.setOnClickListener(this)
+        }
+
+        fun bind(resolveInfo: ResolveInfo) {
+            this.resolveInfo = resolveInfo
+            val packageManager = itemView.context.packageManager
```

Comment:
The adapter should do as little processing of data as possible when binding data to the UI.
For this reason I propose to pass the appName alongside `ResolveInfo` as entry to the adapter.
We still need the `ResolveInfo` to be used when an item is clicked, that's why we cannot get rid of it.

Reply:
You can use a `Pair` to couple the data